src/types.rs

```diff
@@ -1410,9 +1410,9 @@ impl ReadRequest {
         let limit: Option<api::ReadLimit> = match limit {
             None => None,
             Some(limit) => Some({
-                if limit.count > 1000 {
+                if limit.count > Some(1000) {
                     Err("read limit: count must not exceed 1000 for unary request")
-                } else if limit.bytes > (1024 * 1024) {
+                } else if limit.bytes > Some(1024 * 1024) {
```

Comment:
I think there's a constant in the file we can use instead of `1024 * 1024` here.